src/commands/lightning/dev/site.ts

```diff
@@ -55,26 +60,27 @@ export default class LightningDevSite extends SfCommand<void> {
       const selectedSite = new ExperienceSite(org, siteName);
       let siteZip: string | undefined;
 
-      if (!selectedSite.isSiteSetup()) {
-        this.log(`[local-dev] initializing: ${siteName}`);
+      // If the site is not setup / is not based on the current release / or get-latest is requested ->
+      // generate and download a new site bundle from the org based on latest builder metadata
+      if (!selectedSite.isSiteSetup() || getLatest) {
```

Comment:
Published sites are no longer tracked in an orgs static resources, so we will always use our local cache of initialized sites unless the user specifically requests the latest version of a site from the org